sdk/freestanding-execution-engine/src/main.rs

```diff
@@ -186,28 +178,30 @@ fn parse_command_line() -> Result<CommandLineOptions, Box<dyn Error>> {
         Vec::new()
     };
 
-    let dump_stdout = if let Some(dump_stdout) = matches.value_of("dump-stdout") {
+    let dump_stdout = {
+        let default_dump_stdout = DEFAULT_DUMP_STDOUT.to_string();
```

Comment:
I don't think this `let` binding is required.  You can just inline `DEFAULT_DUMP_STDOUT.to_string()`.

Reply:
If I inline it in the obvious way I get:
```
error[E0716]: temporary value dropped while borrowed
   --> src/main.rs:184:25
    |
184 |             .unwrap_or(&DEFAULT_DUMP_STDOUT.to_string());
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ - temporary value is freed at the end of this statement
    |                         |
    |                         creates a temporary which is freed while still in use
185 |         if let Ok(dump_stdout) = bool::from_str(dump_stdout) {
    |                                                 ----------- borrow later used here
    |
    = note: consider using a `let` binding to create a longer lived value
```

There might be a less obvious way of inlining it.

Reply:
I think you can just do `unwrap_or(DEFAULT_DUMP_STDOUT)`.  The constant is already of type `&str`, which is then converted to `String`, and then a slice is taken again to get another `&str`.

Reply:
`DEFAULT_DUMP_STDOUT` is currently a `bool`. But I could change it into a `&'static str`. Then that would work.

Reply:
Ah sorry, yeah, I'm causing confusion.  Yes, I think `DEFAULT_DUMP_STDOUT` should be a `&str` as we only ever work with it in string form, anyway, and never rely on it being of type `bool`, as far as I can see.

Reply:
I've pushed an update.